lib/util/strmap.ts

```diff
@@ -0,0 +1,80 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+export = StrMap;
+
+class StrMap<T> {
+    private _size: number = 0;
+    private _map: {[key: string]: T} = {};
+
+    get size(): number {
+        return this._size;
+    }
+
+    clear(): void {
+        this._size = 0;
+        this._map = {};
+    }
+
+    set(key: string, val: T): void {
+        var mkey = mangle(key);
+        if (!(mkey in this._map)) {
+           ++this._size;
+        }
+        this._map[mkey] = val;
+    }
+
+    get(key: string): T {
+        return this._map[mangle(key)];
+    }
+
+    has(key: string): boolean {
+        return mangle(key) in this._map;
```

Comment:
I would wrap the expression in parenthesis for clarity here to emphasize the boolean result is being returned.